File: app-presentation/src/main/java/com/zxjdev/smile/presentation/application/util/image/ImageLoader.java
package com.zxjdev.smile.presentation.application.util.image;

import android.support.annotation.DrawableRes;
import android.widget.ImageView;

public interface ImageLoader {

  void loadImage(String url, ImageView imageView);

  void loadImage(@DrawableRes int res, ImageView imageView);

  void loadCircleImage(String url, ImageView imageView);

  void loadCircleImage(@DrawableRes int res, ImageView imageView);
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/common/util/image/ImageLoader.kt
package com.zxjdev.smile.presentation.common.util.image

import androidx.annotation.DrawableRes
import android.widget.ImageView

interface ImageLoader {

    fun loadImage(url: String, imageView: ImageView)

    fun loadImage(@DrawableRes res: Int, imageView: ImageView)

    fun loadCircleImage(url: String?, imageView: ImageView)

    fun loadCircleImage(@DrawableRes res: Int, imageView: ImageView)
}
